File: scripts/urlencode.rs
use std::io;

fn urlencode(s: &str) -> String {
    let mut encoded = String::new();
    for c in s.chars() {
        if c.is_ascii_alphanumeric() || ".~-_".contains(c) {
            encoded.push(c);
        } else {
            encoded.push_str(&format!("%{:02X}", c as u8));
        }
    }
    encoded
}

fn main() {
    println!("Enter the string you want to URL encode:");
    let mut input_string = String::new();
    io::stdin().read_line(&mut input_string).expect("Failed to read input");
    let input_string = input_string.trim_end(); // Remove trailing newline character
    
    let encoded_string = urlencode(input_string);
    println!("{}", encoded_string);
}

File: scripts/urlencode.java
import java.util.Scanner;

public class URLEncoder {

    public static String urlencode(String s) {
        StringBuilder encoded = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (Character.isLetterOrDigit(c) || ".~-_".indexOf(c) != -1) {
                encoded.append(c);
            } else {
                encoded.append(String.format("%%%02X", (int) c));
            }
        }
        return encoded.toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter the string you want to URL encode:");
        String inputString = scanner.nextLine();
        String encodedString = urlencode(inputString);
        System.out.println(encodedString);
        scanner.close();
    }
}
